app/Http/Controllers/RoleController.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\User;
+use Illuminate\Validation\Rule;
+use Illuminate\Http\Request;
+use Spatie\Permission\Models\Role;
+use Spatie\Permission\Models\Permission;
+
+class RoleController extends Controller
+{
+    public function __construct()
+    {
+        $this->middleware('role:admin');
+    }
+    
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $roles = Role::all();
+        return response()->json(['status' => 'success', 'roles' => $roles]);
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $this->validate($request, [
+            'name'=>'required|unique:roles',
+        ]);
+        
+        $role = new Role();
+        $role->name = $request->input('name');
+        $role->save();
+        
+        if ($request->has('permissions')) {
+            foreach ($request->input('permissions') as $permission) {
+                $dbPerm = Permission::find($permission);
+                if ($dbPerm) {
+                    $role->givePermissionTo($dbPerm);
```

Comment:
`$role->givePermissionTo($dbPerm)` can accept a string, a `\Spatie\Permission\Models\Role object` or an `\Illuminate\Support\Collection` object, so we can get rid of the `foreach`

Reply:
The foreach was deliberately used to allow for returning an error message that's actually useful to the client in case they specify an incorrect permission name. I can remove it, but your error messages may suck as a result.